src/mpid/ch4/src/ch4_vci.h

```diff
@@ -94,7 +94,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_get_sender_vci(MPIR_Comm * comm,
 #if MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__VCI
     MPIR_Assert(comm);
     /* TODO: implement implicit hashing using other parameters */
-    return comm->seq;
+    return comm->seq % MPIDI_global.n_vcis;
```

Comment:
VCI from ch4 are (should) never be directly used. Netmod always need apply the mod, reference `MPIDI_OFI_get_vni`. The reason is that netmod may have different number of vnis than `MPIDI_global.n_vcis`. So the netmod mod is necessary, and the mod here is unnecessary. The mod is expensive, and should try to avoid.